runtime/compiler/control/JitDump.cpp

```diff
@@ -483,9 +483,84 @@ runJitdump(char *label, J9RASdumpContext *context, J9RASdumpAgent *agent)
 
    try
       {
-      // if our compinfo is null, we are an application thread
+      // Process user defined options first. The user can specify any method signature to be recompiled at an arbitrary
+      // dump event. This is typically used for debugging purposes so we process this option first if it exists.
+      if (NULL != agent->dumpOptions)
+         {
+         const char *indexOfDot = strchr(agent->dumpOptions, '.');
+         const char *indexOfParen = strchr(agent->dumpOptions, '(');
+
+         if ((NULL != indexOfDot) && (NULL != indexOfParen))
+            {
+            TR::VMAccessCriticalSection findUserMethod(TR_J9VMBase::get(jitConfig, crashedThread));
+
+            const char *className = agent->dumpOptions;
+            U_32 classNameLength = static_cast<U_32>(indexOfDot - agent->dumpOptions);
+            const char *methodName = indexOfDot + 1;
+            U_32 methodNameLength = static_cast<U_32>(indexOfParen - methodName);
+            const char *methodSig = indexOfParen;
+            U_32 methodSigLength = strlen(methodSig);
+
+            J9MethodFromSignatureWalkState state;
+            J9Method *userMethod = allMethodsFromSignatureStartDo(&state, context->javaVM, 0, className, classNameLength, methodName, methodNameLength, methodSig, methodSigLength);
+            while (NULL != userMethod)
+               {
+               J9JITExceptionTable *metadata = jitConfig->jitGetExceptionTableFromPC(crashedThread, reinterpret_cast<UDATA>(userMethod->extra));
+               if (NULL != metadata)
+                  {
+                  auto *bodyInfo = reinterpret_cast<TR_PersistentJittedBodyInfo *>(metadata->bodyInfo);
+                  if (NULL != bodyInfo)
+                     {
+                     recompilationThreadInfo->resumeCompilationThread();
```

Comment:
Out of curiosity, does this need to happen inside the loop? What happens if it's only done once? The diag thread doesn't self-suspend right?

Reply:
It doesn't need to happen in the loop per say. Once the diagnostic thread is activated, it will not self suspend. Having the `resumeCompilationThread` after all the checks is necessary though. For example if no method was found with the signature provided we don't really want to resume the diagnostic thread.

I suppose we could just preemptively do it even if no signature matches. I'm ok with either or. Let me know which we should prefer and I can make the change.

Reply:
I don't think it matters too much right now; maybe it could be put into another PR if there's going to be any kind of final clean up. FWIW though, in the other places, the call to `resumeCompilationThread` is guarded by a check to see if either it or the recompilationThread is not NULL; also in one place we print to the vlog and in another we don'; so maybe all of that can just be addressed in a future clean up PR.